File: problems/hackerrank/Tutorials/30 Days of Code/Day 2: Operators/solution.cpp
#include <cmath>
#include <iostream>
using namespace std;


float total_cost(float meal_cost, float tip_percent, float tax_percent) {
    float tip = meal_cost * tip_percent / 100;
    float tax = meal_cost * tax_percent / 100;
    return meal_cost + tip + tax;
}


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    float meal_cost;
    float tip_percent;
    float tax_percent;

    cin >> meal_cost;
    cin >> tip_percent;
    cin >> tax_percent;
    float cost = total_cost(meal_cost, tip_percent, tax_percent);
    cout << "The total meal cost is " << round(cost) << " dollars.";


    return 0;
}

File: problems/hackerrank/Tutorials/30 Days of Code/Day 2: Operators/solution.clj
(defn tip [meal-cost tip-percent]
    (* meal-cost (double (/ tip-percent 100))))

(defn tax [meal-cost tax-percent]
    (* meal-cost (double (/ tax-percent 100))))

(defn cost [meal-cost tip-percent tax-percent]
    (Math/round
        (+ meal-cost
           (tip meal-cost tip-percent)
           (tax meal-cost tax-percent))))

(let [
    meal-cost (Double/parseDouble (read-line))
    tip-percent (Double/parseDouble (read-line))
    tax-percent (Double/parseDouble (read-line))
    cost (cost meal-cost tip-percent tax-percent)
    ]
    (println (str "The total meal cost is " cost " dollars.")))
